### PR TITLE
first stab at implementation the LDAP AuthProvider

### PR DESCRIPTION
**Pull request by stephanebastian**
implemented #325
This implementation does authentication via LDAP. 
Note that it doesn't do authorizatiopn via LDAP. However, it's not a regression since the current LDAP provider (based on Shiro) doesn't seem to support LDAP authorization either.
@pmlopes Could you please review ?

### CODE REVIEW

File: vertx-auth-ldap/src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+== The property file auth provider
```
Comment (pmlopes) [inline vertx-auth-ldap/src/main/asciidoc/index.adoc]: I think this is a copy paste error, the doc is for the properties file not ldap.

File: vertx-auth-ldap/src/main/java/io/vertx/ext/auth/ldap/LdapAuthentication.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+public interface LdapAuthentication extends AuthProvider {
+
+  /**
+   * Create a LDAP authentication provider
+   * 
+   * @param vertx  the Vert.x instance
+   * @param options  the ldap options
+   * @return  the authentication provider
+   */
+  @GenIgnore
```
Comment (pmlopes) [inline vertx-auth-ldap/src/main/java/io/vertx/ext/auth/ldap/LdapAuthentication.java]: `options` is a data object this should not be needed.

**Comment by pmlopes** [comment]
This is great and we can merge once the 2 minor points have been addressed :+1:

File: vertx-auth-ldap/src/main/java/io/vertx/ext/auth/ldap/LdapAuthentication.java (hunk 2 above)
Comment (stephanebastian) [reply vertx-auth-ldap/src/main/java/io/vertx/ext/auth/ldap/LdapAuthentication.java]: What do you mean ? not sure I understand. 
Right now, the options is a mandatory parameter to configure the LDAP provider

File: vertx-auth-ldap/src/main/asciidoc/index.adoc (hunk 1 above)
Comment (stephanebastian) [reply vertx-auth-ldap/src/main/asciidoc/index.adoc]: yep definitely a cut/paste ;) I modified its content, let me know if it's ok

File: vertx-auth-ldap/src/main/java/io/vertx/ext/auth/ldap/LdapAuthentication.java (hunk 2 above)
Comment (stephanebastian) [reply vertx-auth-ldap/src/main/java/io/vertx/ext/auth/ldap/LdapAuthentication.java]: oh got it, didn't see the GenIgnore.... Let me fix it right away

**Pull request merged**